--- mpich-1.2.1/mpid/ch_p4/Makefile.in.orig	Wed Aug 16 21:22:39 2000
+++ mpich-1.2.1/mpid/ch_p4/Makefile.in	Tue May 29 22:11:11 2001
@@ -31,7 +31,7 @@
 # found in, not the directory where the compiler is running (!)
 # Trailing / causes problems for some versions of gcc (was -I./)
 # Was -I../.. -I../ch_p4
-CFLAGS   = @CFLAGS@ -I${top_srcdir}/mpid/ch2 -I${srcdir} -I${top_srcdir}/mpid/util -Ip4 -Ip4/include
+CFLAGS   = @CFLAGS@ ${OPTFLAGS} -I${top_srcdir}/mpid/ch2 -I${srcdir} -I${top_srcdir}/mpid/util -Ip4 -Ip4/include
 
 # See mpid/ch_adi2/Makefile.in for a full list
 SOURCE   = adi2recv.c adi2send.c adi2init.c adi2probe.c adi2ssend.c \
--- mpich-1.2.5/src/env/Makefile.in.orig	Tue Sep  5 16:23:17 2000
+++ mpich-1.2.5/src/env/Makefile.in	Tue May 29 22:28:29 2001
@@ -116,10 +116,10 @@
 # the MPI libraries will work
 msgqdllloc.o: ${top_srcdir}/src/env/msgqdllloc.c
 	@if test -n "@SHAREDKIND_FOR_TV@" -a "@SHAREDKIND_FOR_TV@" != "ignore" ; then \
-	  ${CC} ${DEFS} ${CFLAGS} -c @CC_SHARED_OPT_FOR_TV@ \
+	  ${CC} ${DEFS} ${CFLAGS} ${OPTFLAGSC} ${OPTFLAGS} -c @CC_SHARED_OPT_FOR_TV@ \
 	  -DMPICH_INFODLL_LOC='"@libdir@/libtvmpich.so"' ${top_srcdir}/src/env/msgqdllloc.c ; \
 	else \
-	    ${CC} ${DEFS} ${CFLAGS} -c ${top_srcdir}/src/env/msgqdllloc.c ; \
+	    ${CC} ${DEFS} ${CFLAGS} ${OPTFLAGSC} ${OPTFLAGS} -c ${top_srcdir}/src/env/msgqdllloc.c ; \
 	fi
 
 .c.o:
--- mpich-1.2.5/src/infoexport/Makefile.in.orig	Fri Jan 14 21:58:04 2000
+++ mpich-1.2.5/src/infoexport/Makefile.in	Tue May 29 22:50:47 2001
@@ -7,7 +7,7 @@
 
 # Define FOR_MPI2=1 to enable some MPI-2 support
 DEFS        = @DEFS@ ${MPIPROFILE} -I. -I${srcdir} @mpich_includes@ -DFOR_MPI2=0
-CFLAGS      = -g @CC_SHARED_OPT_FOR_TV@ @CFLAGS@
+CFLAGS      = @OPTFLAGS@ @CC_SHARED_OPT_FOR_TV@ @CFLAGS@
 
 @VPATH@
 
--- mpich-1.2.1/src/fortran/configure.in.orig	Tue Aug 29 23:53:00 2000
+++ mpich-1.2.1/src/fortran/configure.in	Tue May 29 22:56:52 2001
@@ -21,7 +21,7 @@
 AC_SUBST(COPTIONS)
 AC_ARG_ENABLE(strict,
 [--enable-strict  - Turn on strict compilation testing when using gcc],
-COPTIONS="${COPTIONS} -Wall -O -Wstrict-prototypes -Wmissing-prototypes -DGCC_WALL")
+COPTIONS="${COPTIONS} -Wall -Wstrict-prototypes -Wmissing-prototypes -DGCC_WALL")
 
 AC_ARG_ENABLE(echo,
 [--enable-echo    - Turn on echoing from within the configure script],
--- mpich-1.2.1/src/fortran/configure.orig	Tue Sep  5 22:05:39 2000
+++ mpich-1.2.1/src/fortran/configure	Tue May 29 23:16:32 2001
@@ -626,7 +626,7 @@
 # Check whether --enable-strict or --disable-strict was given.
 if test "${enable_strict+set}" = set; then
   enableval="$enable_strict"
-  COPTIONS="${COPTIONS} -Wall -O -Wstrict-prototypes -Wmissing-prototypes -DGCC_WALL"
+  COPTIONS="${COPTIONS} -Wall -Wstrict-prototypes -Wmissing-prototypes -DGCC_WALL"
 fi
 
 
--- mpich-1.2.5.3/configure.in.orig	2004-04-05 15:48:47.000000000 +0200
+++ mpich-1.2.5.3/configure.in	2004-07-04 13:39:28.170198120 +0200
@@ -3510,6 +3510,8 @@
   export TESTCC ;\
   export TESTF77 ;\
   export TESTF90 ;\
+  COPTIONS="$OPTFLAGSC $OPTFLAGS"; export COPTIONS;\
+  FOPTIONS="$OPTFLAGSF $OPTFLAGS"; export FOPTIONS;\
   export mpich_includes ;\
   export LIB_PATH ;\
   export BASE_LIB_LIST ;\
@@ -3684,6 +3686,7 @@
   export FROM_MPICH;\
   export int_len ; export void_star_len ; export OFFSET_KIND ; \
   export long_long_len ; \
+  CFLAGS="$OPTFLAGSC $OPTFLAGS" ; export CFLAGS ; \
   "${top_srcdir}/romio/configure" $ROMIO_ARGS $ROMIO_FILE_SYSTEM \
             --with-mpi=mpich $romio_config_args $enable_args $with_args )
   rc=$?
@@ -3773,6 +3776,7 @@
                                   --prefix=${prefix} $enable_args $with_args \
                                   --libdir=${libdir} --bindir=${bindir} \
 				  --includedir=${includedir} \
+				  --with-cflags="$OPTFLAGSC $OPTFLAGS" \
 				  $MPE_ARGS )
     if test $? = 0 ; then
         echo "Done configuring MPE Profiling Libraries" ; echo
--- mpich-1.2.5.3/configure.orig	2004-04-07 19:26:01.000000000 +0200
+++ mpich-1.2.5.3/configure	2004-07-04 13:40:20.934176776 +0200
@@ -15634,6 +15634,8 @@
   export TESTCC ;\
   export TESTF77 ;\
   export TESTF90 ;\
+  COPTIONS="$OPTFLAGSC $OPTFLAGS"; export COPTIONS;\
+  FOPTIONS="$OPTFLAGSF $OPTFLAGS"; export FOPTIONS;\
   export mpich_includes ;\
   export LIB_PATH ;\
   export BASE_LIB_LIST ;\
@@ -15900,6 +15902,7 @@
   export FROM_MPICH;\
   export int_len ; export void_star_len ; export OFFSET_KIND ; \
   export long_long_len ; \
+  CFLAGS="$OPTFLAGSC $OPTFLAGS" ; export CFLAGS ; \
   "${top_srcdir}/romio/configure" $ROMIO_ARGS $ROMIO_FILE_SYSTEM \
             --with-mpi=mpich $romio_config_args $enable_args $with_args )
   rc=$?
@@ -15994,6 +15997,7 @@
                                   --prefix=${prefix} $enable_args $with_args \
                                   --libdir=${libdir} --bindir=${bindir} \
 				  --includedir=${includedir} \
+				  --with-cflags="$OPTFLAGSC $OPTFLAGS" \
 				  $MPE_ARGS )
     if test $? = 0 ; then
         echo "Done configuring MPE Profiling Libraries" ; echo
--- mpich-1.2.5.3/Makefile.in.orig	Tue Sep  5 22:06:05 2000
+++ mpich-1.2.5.3/Makefile.in	Wed May 30 01:07:43 2001
@@ -327,7 +327,7 @@
 	CC="@CC@" ; export CC ; \
 	ldir=`echo $(libdir) | sed -e 's%$(MPIR_HOME)/%%g'` ; \
 	if ${top_srcdir}/${CPP_DIR}/configure --with-mpich=.. \
-			--with-libopt=" " \
+			--with-libopt="@OPTFLAGS@" \
 			 --with-profiling \
 			 --with-mpichbin=${binbuild_dir} \
 			 --with-mpichlib=${libbuild_dir} \
@@ -467,6 +467,7 @@
 serv_p4 server:
 	@if [ ! -d mpid/server ] ; then mkdir mpid/server ; fi
 	@( cd mpid/server && rm -f config.cache ; \
+	  CFLAGS="@OPTFLAGSC@ @OPTFLAGS@"; export CFLAGS ; \
 	  CC="@CC@" ; export CC ; MAKE="${MAKE}" ; export MAKE ; \
 	  ${top_srcdir}/mpid/server/configure ; $(MAKE) serv_p4 ; \
 	  cp serv_p4 $(binbuild_dir) )
