--- mpich-3.1.3/configure.ac.orig	2014-11-22 21:36:39.473469954 +0100
+++ mpich-3.1.3/configure.ac	2014-11-22 22:20:53.770025233 +0100
@@ -1212,6 +1212,8 @@
 AC_SUBST([opasrcdir])
 opalibdir=""
 AC_SUBST([opalibdir])
+opalibopt=""
+AC_SUBST([opalibopt])
 opalib=""
 AC_SUBST([opalib])
 
@@ -1219,6 +1221,7 @@
     if test -e "${use_top_srcdir}/src/openpa" ; then
         opasrcdir="${master_top_builddir}/src/openpa"
         opalibdir="${master_top_builddir}/src/openpa/src"
+        opalibopt="-L${master_top_builddir}/src/openpa/src"
         opalib="${master_top_builddir}/src/openpa/src/lib${OPALIBNAME}.la"
         PAC_APPEND_FLAG([-I${use_top_srcdir}/src/openpa/src],[CPPFLAGS])
         PAC_APPEND_FLAG([-I${master_top_builddir}/src/openpa/src],[CPPFLAGS])
@@ -1255,8 +1258,10 @@
     if test -d ${with_openpa_prefix}/lib64 ; then
         PAC_APPEND_FLAG([-L${with_openpa_prefix}/lib64],[WRAPPER_LDFLAGS])
         opalibdir="${with_openpa_prefix}/lib64"
+        opalibopt="-L${with_openpa_prefix}/lib64"
     else
         opalibdir="${with_openpa_prefix}/lib"
+        opalibopt="-L${with_openpa_prefix}/lib"
     fi
     PAC_APPEND_FLAG([-L${with_openpa_prefix}/lib],[WRAPPER_LDFLAGS])
 fi
--- mpich-3.1.3/Makefile.am.orig	2014-10-07 23:50:23.000000000 +0200
+++ mpich-3.1.3/Makefile.am	2014-11-22 22:21:28.990023755 +0100
@@ -48,7 +48,7 @@
 errnames_txt_files = 
 
 external_subdirs = @mplsrcdir@ @opasrcdir@
-external_ldflags = -L@mpllibdir@ -L@opalibdir@
+external_ldflags = -L@mpllibdir@ @opalibopt@
 external_libs = @EXTERNAL_LIBS@
 mpi_convenience_libs =
 pmpi_convenience_libs = @mpllib@ @opalib@
--- mpich-3.1.3/examples/Makefile.am.orig	2014-10-07 23:50:23.000000000 +0200
+++ mpich-3.1.3/examples/Makefile.am	2014-11-22 23:06:01.823244920 +0100
@@ -36,7 +36,7 @@
 # the make-time instances of libpmpi.la and libmpi.la live here
 AM_LDFLAGS += -L../lib
 # the make-time instances of libmpl.la and libopa.la live here
-AM_LDFLAGS += -L@mpllibdir@ -L@opalibdir@
+AM_LDFLAGS += -L@mpllibdir@ @opalibopt@
 
 # Wrapper LDFLAGS need to be added at the end to make sure we link
 # with the libraries we just built, and not any previously installed
