--- mpich-1.2.5.3/mpid/ch_p4/configure.orig	2003-08-12 15:19:25.000000000 +0200
+++ mpich-1.2.5.3/mpid/ch_p4/configure	2004-07-04 12:36:33.739998728 +0200
@@ -1186,22 +1186,8 @@
 fi
 
         if test -n "$RSHCOMMAND" ; then
-	    echo $ac_n "checking whether $RSHCOMMAND works""... $ac_c" 1>&1
-echo "configure:1191: checking whether $RSHCOMMAND works" >&2
-            output=`$RSHCOMMAND $machine -n true 2>&1 </dev/null`
-            if test -n "$output" ; then
-		echo "$ac_t""no" 1>&1
-		echo "Errors while trying to run true on $machine with $program"
-		echo "$output"
-		RSHCOMMAND=""
-		# HORRIBLE, TERRIBLE BUG in AUTOCONF!!!
-		# It is determined to INCORRECTLY remember some results,
-		# while forgetting others (like "cross-compiling")
-		unset ac_cv_path_RSHCOMMAND
-            else
 		echo "$ac_t""yes" 1>&1
 		break
-	    fi
         fi
     done
     # Check ssh with special options to avoid user prompting which causes the
@@ -1243,22 +1229,7 @@
 fi
 
     	if test -n "$RSHCOMMAND" ; then
-    	    echo $ac_n "checking whether $RSHCOMMAND works""... $ac_c" 1>&1
-echo "configure:1248: checking whether $RSHCOMMAND works" >&2
-    	    output=`$RSHCOMMAND -o "StrictHostKeyChecking no" \
-    	      -o "BatchMode yes" $machine -n true 2>&1 </dev/null`
-    	    if test $? -ne 0 ; then
-    		echo "$ac_t""no" 1>&1
-    		echo "Errors while trying to run true on $machine with ssh"
-    		echo "$output"
-    		RSHCOMMAND=""
-    		# HORRIBLE, TERRIBLE BUG in AUTOCONF!!!
-    		# It is determined to INCORRECTLY remember some results,
-    		# while forgetting others (like "cross-compiling")
-    		unset ac_cv_path_RSHCOMMAND
-    	    else
     		echo "$ac_t""yes" 1>&1
-    	    fi
     	fi
     fi
 fi
