--- mpich-1.2.7p1/configure.in~	2012-04-04 13:02:28.073692580 +0200
+++ mpich-1.2.7p1/configure.in	2012-04-04 13:55:29.800502795 +0200
@@ -1804,7 +1804,7 @@
             # (the autoconf tests are too GNU-centric)
 	    if test -n "$GXXTRIAL" ; then 
                 cat > conftest.cc <<EOF
-#include <iostream.h>
+#include <iostream>
 EOF
 	        rm -f conftest.out
   	        if $GXXTRIAL $CXXFLAGS -c conftest.cc >conftest.out 2>&1 ; then
@@ -1845,10 +1845,10 @@
     # used above is used only to *find* a compiler; if the user selected
     # a compiler that doesn't work, we still need the test.
     if test -n "$CPP_COMPILER" ; then 
-        AC_MSG_CHECKING([whether selected C++ compiler can compile iostream.h])
+        AC_MSG_CHECKING([whether selected C++ compiler can compile iostream])
         rm -f conftest*
         cat > conftest.cc <<EOF
-#include <iostream.h>	
+#include <iostream>	
 EOF
         if $CPP_COMPILER $CXXFLAGS -c conftest.cc>conftest.out 2>&1 ; then 
             AC_MSG_RESULT(yes)
--- mpich-1.2.7p1/configure~	2012-04-04 13:56:35.643839099 +0200
+++ mpich-1.2.7p1/configure	2012-04-04 13:57:10.133840656 +0200
@@ -2748,7 +2748,7 @@
             # (the autoconf tests are too GNU-centric)
 	    if test -n "$GXXTRIAL" ; then 
                 cat > conftest.cc <<EOF
-#include <iostream.h>
+#include <iostream>
 EOF
 	        rm -f conftest.out
   	        if $GXXTRIAL $CXXFLAGS -c conftest.cc >conftest.out 2>&1 ; then
@@ -2844,13 +2844,13 @@
 
 fi
 if test -z "$ac_echo_test" -a 1 = 1 ; then
-echo $ac_n "checking whether selected C++ compiler can compile iostream.h""... $ac_c"
+echo $ac_n "checking whether selected C++ compiler can compile iostream""... $ac_c"
 else
-echo $ac_n "checking whether selected C++ compiler can compile iostream.h""... $ac_c" 1>&1
+echo $ac_n "checking whether selected C++ compiler can compile iostream""... $ac_c" 1>&1
 fi
         rm -f conftest*
         cat > conftest.cc <<EOF
-#include <iostream.h>	
+#include <iostream>
 EOF
         if $CPP_COMPILER $CXXFLAGS -c conftest.cc>conftest.out 2>&1 ; then 
             if test -z "$ac_echo_test" -a 1 = 1 ; then
